### PR TITLE
Add timeout= parameter to all requests.* calls

### PR DESCRIPTION
**Pull request by EricSoroos**
### Proposed fixes:

Requests will block forever if the remote server hangs and there's no
timeout.

* Two new config variables:
  - ckan.resource_proxy.timeout -- specifically for resource_proxy
  - ckan.requests.timeout -- for all other uses of requests in the
  code base.
* All timeouts currently defaulting to 10 sec

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [x] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by amercader** [comment]
@EricSoroos do you want to clean this up and address the comments so it can get in the next patch release?

File: ckan/lib/captcha.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # encoding: utf-8
 
-from ckan.common import config
+from ckan.common import config, asint
 
 import requests
 
+TIMEOUT = asint(config.get('ckan.requests.timeout', 10))
```
[the remark this answers is not in the document]
Comment (EricSoroos) [reply ckan/lib/captcha.py:7]: Are you saying that I should add `config.set_default('ckan.requests.timeout', 5)` in the update_config method?

[the remark this answers is not in the document]
Comment (EricSoroos) [reply ckan/lib/captcha.py:7]: And how does that interact with the new config_declaration.yaml?  Does that actually apply at runtime if there's a default value?

**Comment by EricSoroos** [comment]
Unclear if the test failure is mine.

**Comment by amercader** [comment]
Thanks @EricSoroos.

I've adapted and cherry-picked this to 2.9 (https://github.com/ckan/ckan/commit/fd2da1b0f73184bb04e4c9fd0d0e9188f194f8fd) and 2.8 (https://github.com/ckan/ckan/commit/c031752113092b3632be73d5403201b602ed9ce8), would be great if you could do a quick review of those. cheers!

**Comment by EricSoroos** [comment]
The backports look ok to me.

**Comment by kowh-ai** [comment]
I think the "Document new options" commit: https://github.com/ckan/ckan/commit/81177ed9b0221ac8412da89e9661795f1fe65322 needs to be merged into **master** as the tests seem to be failing at the moment...the new `ckan.requests.timeout` label is referenced by CHANGELOG.rst but doesn't exist in the configuration.rst file. CKAN 2.9 and 2.8 are fine.

**Pull request merged**